public/app/listeners/loadDataset.js

```diff
@@ -24,29 +25,35 @@ const createNewDataset = ({ name, filepath }) => {
     id: fileId,
     name,
     filepath: destPath,
+    description,
     size: sizeInKiloBytes,
     createdAt,
     lastModified,
   };
 };
 
 const loadDataset = (mainWindow, db) => async (event, args) => {
-  const { fileLocation } = args;
-
-  // filename without extension
-  const filename = path
-    .basename(fileLocation)
-    .slice(0, -path.extname(fileLocation).length);
+  const { fileLocation, fileDescription } = args;
+  let { fileCustomName } = args;
```

Comment:
I am not sure it is a good practice to edit `fileCustomName` of `args`.
Maybe consider 
```
const { fileLocation, ...., fileCustomName} = args;
const defaultFileName = path.basename.... 
// or without this tmp variable to avoid computing it if fileCustomName exists
const fileName = fileCustomName || defaultFileName

createNewDataset({
name: fileName
....
}) 
```

Reply:
nice! makes sense